doc/man/man3/crypto_lock_init.3monocypher

```diff
@@ -64,43 +64,42 @@ and
 .Xr crypto_aead_unlock 3monocypher .
 Prefer those simpler functions if possible.
 .Pp
-This incremental interface allows splitting up decryption and
-authentication of messages too large to fit into a single buffer
-or to handle individual pieces of data located in different buffers.
-The arguments are the same as described for the direct interface on
+This incremental interface can be used to encrypt and decrypt messages too
+large to fit in a single buffer.
+The arguments are the same as described for the direct interface described in
 .Xr crypto_lock 3monocypher .
 .Pp
-This interface uses four steps:
+Encryption requires four steps:
 .Bl -bullet
 .It
-Initialisation with
+Initialise a context with
 .Fn crypto_lock_init .
-This sets up a context for encryption or decryption.
-The same function is used for both.
 .It
-Authentication with
+Authenticate additional data, if any, with
 .Fn crypto_lock_auth .
-This authenticates (or verifies) additional data, if any.
 .It
-Update with
-.Fn crypto_lock_update
-for encryption, or
-.Fn crypto_unlock_update
-for decryption.
-This encrypts (or decrypts) and authenticates (or verifies) part of
-the secret message.
+Encrypt and authenticate some data with
+.Fn crypto_lock_update .
 .It
-Finalisation with
-.Fn crypto_lock_final
-to generate the MAC, or
-.Fn crypto_unlock_final
-to verify the MAC.
+Generate the MAC with
+.Fn crypto_lock_final .
 .El
 .Pp
-Because the incremental interface interleaves encryption and
-authentication, it treats corrupted messages three times slower than
-the direct interface.
-Users who expect a high corruption rate may want a different approach.
```

Comment:
If you think this should be kept in I can put it back and rephrase it a little

Reply:
Yes, this should be kept. It was quite a long struggle to get it in because [the incremental interface itself was quite controversial](https://github.com/LoupVaillant/Monocypher/issues/29) precisely *because* of this speed tradeoff.

Reply:
I stuck it in an implementation details block at the end

it doesn't seem super important to me

Reply:
I agree with putting that paragraph in the implementation details section.

By the way, I learned about another, much more important, reason why incremental processing is controversial: the possibility of trusting unauthenticated messages.

As such, incremental authentication (or signature verification for that matter) isn't problematic. What is is the idea of processing *unauthenticated* data further down the processing pipeline. And if the message turned out to be a forgery, such processing must be cleanly undone, and mustn’t have triggered any side effects, perhaps even timed responses that may inform the attacker about the contents of the message.

It's a security nightmare.

Whatever the user does, she must not trust the message before the authentication code has been verified. In practice, this generally means simply dumping the message to disk, and ignoring (or erasing) it if it's a forgery.

I think we should add a "security considerations" section for this.